src/google/protobuf/util/field_mask_util.cc

```diff
@@ -332,14 +331,14 @@ void FieldMaskTree::AddPath(const std::string& path) {
   }
   bool new_branch = false;
   Node* node = &root_;
-  for (int i = 0; i < parts.size(); ++i) {
+  for (const std::string& part : parts) {
     if (!new_branch && node != &root_ && node->children.empty()) {
       // Path matches an existing leaf node. This means the path is already
       // covered by this tree (for example, adding "foo.bar.baz" to a tree
       // which already contains "foo.bar").
       return;
     }
-    const std::string& node_name = parts[i];
+    const std::string& node_name = part;
```

Comment:
I think this loop could do the same thing as the one below and just name the loop variable `node_name` so that this line can be deleted.

Reply:
Done